### PR TITLE
First pass at displaying hustings information on a postcode page.

### PR DESCRIPTION
**Pull request by davidmiller**
What this doesn't do is: 
```Should only need to add to the
"select_related" section of PostcodeToPostsMixin.postcode_to_posts if
it's a FK
```

I'm still not entirely sure I understand how the data model works here, but the select_related incantations I attempted resulted in errors :( 

I would have thought that e.g.
`pes = pes.select_related('post__husting')` would work - but presumably am missing something blindingly obvious.

Also: I am not attempting to pull data directly from a google spreadsheet, but from a local CSV of page 3 of that spreadsheet if that works...

### CODE REVIEW

File: wcivf/apps/elections/templates/elections/postcode_view.html [hunk 1]
```diff
@@ -99,6 +99,36 @@
 
 
   </div>
+
+  {% if postelection.post.husting_set.count %}
```
Comment (symroe) [inline wcivf/apps/elections/templates/elections/postcode_view.html]: I think `exists` is better here than `count`

File: wcivf/apps/elections/templates/elections/postcode_view.html [hunk 2]
```diff
@@ -99,6 +99,36 @@
...
+    <div class="card">
+      <h3>Hustings</h3>
+      {% for husting in postelection.post.husting_set.all %}
```
Comment (symroe) [inline wcivf/apps/elections/templates/elections/postcode_view.html]: Do we care about splitting these up in to "past" and "future" lists?

File: wcivf/apps/hustings/management/commands/import_hustings.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                starts, data.start_time
+            )
+        if data.end_time:
+            ends = dt_from_string(data.date)
+            ends = set_time_string_on_datetime(
+                ends, data.end_time
+            )
+        # This seems absurd. Maybe there's a better way to spell this
+        # in the ORM ? Maybe I don't understand the data model properly?
+        election = Election.objects.get(slug=data.electionid)
```
Comment (symroe) [inline wcivf/apps/hustings/management/commands/import_hustings.py]: Sorry, my fault. I think the FK needs to be on `PostElection` not `Post`. `Post` is the area, `PostElection` is the place with an election. That will make this query simpler and the `select_related` work, I think.

File: wcivf/apps/hustings/models.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from django.db import models
+
+from elections.models import Post
+
+
+class Husting(models.Model):
+    """
+    A Husting.
+    """
+    post    = models.ForeignKey(Post)
```
Comment (symroe) [inline wcivf/apps/hustings/models.py]: See comment above, this should be to `PostElection`

**Comment by tfgg** [comment]
image

File: wcivf/apps/elections/templates/elections/postcode_view.html [hunk 5]
```diff
@@ -99,6 +99,36 @@
...
+        <h4>{{ husting.title}}</h4>
+      <p>
+        {{ husting.starts | date:"D b d P"}}
```
Comment (tfgg) [inline wcivf/apps/elections/templates/elections/postcode_view.html]: Capitalize the month?

File: wcivf/apps/elections/templates/elections/postcode_view.html (hunk 1 above)
Comment (davidmiller) [inline wcivf/apps/elections/templates/elections/postcode_view.html]: Meh, if you say so...

File: wcivf/apps/elections/templates/elections/postcode_view.html (hunk 2 above)
Comment (davidmiller) [inline wcivf/apps/elections/templates/elections/postcode_view.html]: I don't as a first pass, I'd rather have them in.
But quite possibly as a second iteration.

File: wcivf/apps/hustings/management/commands/import_hustings.py (hunk 3 above)
Comment (davidmiller) [inline wcivf/apps/hustings/management/commands/import_hustings.py]: > "Post is the area, PostElection is the place with an election."

I'm not sure I even know what the difference between area and place with an election is.
Names are hard etc...

File: wcivf/apps/elections/templates/elections/postcode_view.html (hunk 5 above)
Comment (davidmiller) [inline wcivf/apps/elections/templates/elections/postcode_view.html]: I don't really understand why Django `| date` doesn't implement B as a filter param. Seems obviously useful to me.

File: wcivf/apps/hustings/management/commands/import_hustings.py (hunk 3 above)
Comment (symroe) [inline wcivf/apps/hustings/management/commands/import_hustings.py]: Post: Cambridge constituency
Election: GE2017
PostElection the election in Cambridge ward (there have been past elections here, so we need one of them for each election).

PostElection is the `through` table of the many to many relationship

**Comment by symroe** [comment]
I think including the migrations are confusing here, since this is a new feature. Can you delete them and just create a single one for this change?

Also, because I can't run the migrations on my computer because of some odd DB problem…

**Comment by davidmiller** [comment]
@symroe squashed

**Comment by symroe** [comment]
Lovely!

**Pull request merged**